### PR TITLE
[2.0.0 beta.6] Licenses & Visibility annotations

### PR DESCRIPTION
**Pull request by St4NNi**
This is a major breaking update that adds licenses to all resources.

## Changelog

- Added License API to create and manage licenses
- Added metadata license and (default)data license tags to all resources
- Added functionality to add Non BelongsTo internal relations on creation
- Update visibility annotations
- Fix typos (plurals)
- Add OpenAPI annotations for Notifications
- Upgrade Rust dependencies

Fixes https://github.com/ArunaStorage/ArunaAPI/pull/98

### CODE REVIEW

File: aruna/api/notification/services/v2/notification_service.proto [hunk 1]
```diff
@@ -7,49 +7,74 @@ option java_package = "com.github.ArunaStorage.java_api.aruna.api.notification.s
...
+  option (google.api.api_visibility).restriction = "SERVER";
 
   // CreateStreamConsumer
   //
   // Creates a new event stream consumer.
   rpc CreateStreamConsumer(CreateStreamConsumerRequest)
-      returns (CreateStreamConsumerResponse) {}
+      returns (CreateStreamConsumerResponse) {
+        option (google.api.http) = {
+          post : "/v2/notifications"
```
Comment (das-Abroxas) [inline aruna/api/notification/services/v2/notification_service.proto:28]:
```suggestion
          post : "/v2/notifications/consumer"
```

File: aruna/api/storage/models/v2/models.proto [hunk 2]
```diff
@@ -342,4 +355,6 @@ message Object {
   repeated DataEndpoint endpoints = 12;
   // Object specific attributes
   repeated Hash hashes = 13;  
+  string metadata_license_tag = 14;
+  string default_data_license_tag = 15;
```
Comment (das-Abroxas) [inline aruna/api/storage/models/v2/models.proto:359]:
```suggestion
  string data_license_tag = 15;
```

File: aruna/api/storage/services/v2/dataset_service.proto [hunk 3]
```diff
@@ -129,15 +143,17 @@ message CreateDatasetRequest {
   string description = 2;
   // dataset specific labels / hooks
   repeated storage.models.v2.KeyValue key_values = 3;
-  // External relations (URLs / IDs from external sources)
-  repeated storage.models.v2.ExternalRelation external_relations = 4;
+  // External / Internal relations (URLs / IDs from external sources)
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/dataset_service.proto:146]:
```suggestion
  // Internal / External relations (URLs / IDs from external sources)
```

File: aruna/api/storage/services/v2/object_service.proto [hunk 4]
```diff
@@ -147,8 +150,8 @@ message CreateObjectRequest {
   string description = 2;
   // collection specific labels / hooks
   repeated storage.models.v2.KeyValue key_values = 3;
-  // External relations (URLs / IDs from external sources)
-  repeated storage.models.v2.ExternalRelation external_relations = 4;
+  // External / Internal relations (URLs / IDs from external sources)
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/object_service.proto:153]:
```suggestion
  // Internal / External relations (URLs / IDs from external sources)
```

File: aruna/api/storage/services/v2/collection_service.proto [hunk 5]
```diff
@@ -130,13 +144,15 @@ message CreateCollectionRequest {
   // collection specific labels / hooks
   repeated storage.models.v2.KeyValue key_values = 3;
   // External relations (URLs / IDs from external sources)
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/collection_service.proto:146]:
```suggestion
  // Internal /External relations (URLs / IDs from external sources)
```

File: aruna/api/storage/services/v2/license_service.proto [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+//
+// Request object for ListLicense
+message ListLicenseRequest {}
+
+// ListLicenseResponse
+//
+// Response object for ListLicense
+message ListLicenseResponse {    
+    repeated storage.models.v2.License licenses = 1;
+}
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/license_service.proto:94]:
```suggestion
// ListLicensesRequest
//
// Request object for ListLicense
message ListLicensesRequest {}

// ListLicenseResponse
//
// Response object for ListLicense
message ListLicensesResponse {    
    repeated storage.models.v2.License licenses = 1;
}
```

File: aruna/api/storage/services/v2/license_service.proto [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    // ListLicense
+    //
+    // Status: BETA
+    //
+    // This returns a list of all licenses
+    rpc ListLicense(ListLicenseRequest) returns (ListLicenseResponse) {
+        option (google.api.http) = {
+        get : "/v2/license"
+        };
+    }
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/license_service.proto:49]:
```suggestion
    // ListLicenses
    //
    // Status: BETA
    //
    // This returns a list of all licenses
    rpc ListLicenses(ListLicensesRequest) returns (ListLicensesResponse) {
        option (google.api.http) = {
        get : "/v2/licenses"
        };
    }
```

File: aruna/api/storage/services/v2/license_service.proto [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+option java_multiple_files = true;
+option java_package = "com.github.ArunaStorage.java_api.aruna.api.storage.services.v2";
+option java_outer_classname = "LicenseService";
+import "aruna/api/storage/models/v2/models.proto";
+import "google/api/annotations.proto";
+import "google/api/visibility.proto";
+
+// AuthorizationService
+//
+// Contains all methods to edit and change user authorization
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/license_service.proto:14]:
```suggestion
// LicenseService
//
// Contains all methods to add, get or list licenses
```

File: aruna/api/storage/services/v2/project_service.proto [hunk 9]
```diff
@@ -157,12 +171,15 @@ message CreateProjectRequest {
   string description = 2;
   // Project specific labels / hooks
   repeated storage.models.v2.KeyValue key_values = 3;
-  // External relations (URLs / IDs from external sources)
-  repeated storage.models.v2.ExternalRelation external_relations = 4;
+  // External / Internal relations (URLs / IDs from external sources)
```
Comment (das-Abroxas) [inline aruna/api/storage/services/v2/project_service.proto:174]:
```suggestion
  // Internal / External relations (URLs / IDs from external sources)
```

**Review by das-Abroxas — COMMENTED** [review]
Made some suggestions to correct comments and naming. They can be addressed, but it wouldn't affect functionality if they were simply rejected.

Overall, it looks good to me and I will approve after the suggestions are applied or rejected :+1:

**Review by lfbrehm — APPROVED** [review]

**Pull request merged**